python/doc/examples/numerical_methods/general_methods/plot_estimate_integral_iterated_quadrature.py

```diff
@@ -1,48 +1,268 @@
+# The domain of integration is defined by the two following functions:
+#
+# .. math::
+#
+#    x \rightarrow u(x) = 2+cos(x)
+#
+# .. math::
+#
+#    x \rightarrow  \ell(x) = -2-cos(x)
+#
+upper_func = ot.SymbolicFunction(["x"], [" 2 + cos(x)"])
+lower_func = ot.SymbolicFunction(["x"], ["-2 - cos(x)"])
 
 # %%
-# Draw the graph of the integrand and the bounds
-g = ot.Graph("Integration nodes", "x", "y", True, "upper right")
-g.add(f.draw([a, a], [b, b]))
-curve = ll[0].draw(a, b).getDrawable(0)
-curve.setLineWidth(2)
-curve.setColor("red")
-g.add(curve)
-curve = u[0].draw(a, b).getDrawable(0)
-curve.setLineWidth(2)
-curve.setColor("red")
-g.add(curve)
-view = viewer.View(g)
+# We draw the iso-lines of the integrand function in the domain of integration.
+# To do that, we define a new function which is the restriction
+# of the integrand function to the integration domain.
+
+
+def kernel3_insideDomain(x):
+    low_x = lower_func([x[0]])[0]
+    up_x = upper_func([x[0]])[0]
+    if x[1] > low_x and x[1] < up_x :
+        return kernel3(x)
+    else:
+        return [0.0]
+
+
+integrand_domain = ot.PythonFunction(2, 1, kernel3_insideDomain)
+a = 2 * m.pi
+b = 4.0
+# here, we ask for more levels than the default number store in :class:`~openturns.ResourceMap`.
+ot.ResourceMap.SetAsUnsignedInteger("Contour-DefaultLevelsNumber", 30)
+g = integrand_domain.draw([-a, -b], [a, b])
+
+# %%
+# We add the bounds of the domain :math:`\set{D}`.
+low_bound = lower_func.draw([-a, -b], [a, b]).getDrawable(0)
+up_bound = upper_func.draw([-a, -b], [a, b]).getDrawable(0)
+low_bound.setColor('red')
+low_bound.setLineWidth(2)
+up_bound.setColor('red')
+up_bound.setLineWidth(2)
+
+g.add(low_bound)
+g.add(up_bound)
+g.setTitle(r'Function $f(x,y) =  cos(2x) * sin(1.5y)$')
+g.setXTitle('x')
+g.setYTitle('y')
+view = otv.View(g)
+
+# %%
+# To get the nodes used to approximate the integral, we need to use a :class:`~openturns.MemoizeFunction`
+# that stores all the calls to the function.
+integrand_memoized = ot.MemoizeFunction(integrand)
+maximumSubIntervals = 4
+maximumError = 1e-4
+GKRule = ot.GaussKronrodRule(ot.GaussKronrodRule.G7K15)
+integration_algo = ot.IteratedQuadrature(ot.GaussKronrod(maximumSubIntervals, maximumError, GKRule))
+I_value = integration_algo.integrate(integrand_memoized, -a, a, [lower_func], [upper_func])
+print('I = ', I_value[0])
+nodes = integrand_memoized.getInputHistory()
+print('Nodes : ', nodes)
+
+# %%
+# We can draw in black the nodes on the graph that contains the iso-lines of the
+# function :math:`f`. We can see that the algorithm focuses on the nodes where
+# the function has rapid variations.
+cloud_nodes = ot.Cloud(nodes)
+cloud_nodes.setPointStyle('dot')
+cloud_nodes.setColor('black')
+
+g.add(cloud_nodes)
+g.setLegendPosition('')
+view = otv.View(g)
 
 # %%
-# compute the integral value
-I2 = ot.IteratedQuadrature().integrate(f, a, b, ll, u)
-print(I2)
-plt.show()
+# Show all the graphs.
+view.ShowAll()
```

Comment:
ResourceMap.Reload too since you're altering it, else it will propagate to other examples

Reply:
ok!